openerp/osv/expression.py

```diff
@@ -858,22 +858,22 @@ def push_result(leaf):
                         push(create_substitution_leaf(leaf, elem, comodel))
                     push(create_substitution_leaf(leaf, AND_OPERATOR, comodel))
 
-            elif len(path) > 1 and column._auto_join:
+            elif len(path) > 1 and column and column._auto_join:
                 raise NotImplementedError('_auto_join attribute not supported on many2many column %s' % left)
 
-            elif len(path) > 1 and column._type == 'many2one':
+            elif len(path) > 1 and column and column._type == 'many2one':
                 right_ids = comodel.search(cr, uid, [(path[1], operator, right)], context=context)
                 leaf.leaf = (path[0], 'in', right_ids)
                 push(leaf)
 
             # Making search easier when there is a left operand as column.o2m or column.m2m
-            elif len(path) > 1 and column._type in ['many2many', 'one2many']:
+            elif len(path) > 1 and column and column._type in ['many2many', 'one2many']:
                 right_ids = comodel.search(cr, uid, [(path[1], operator, right)], context=context)
                 table_ids = model.search(cr, uid, [(path[0], 'in', right_ids)], context=dict(context, active_test=False))
                 leaf.leaf = ('id', 'in', table_ids)
                 push(leaf)
 
-            elif not column:
+            elif not column or field and field.search:
```

Comment:
@hbrunn that extra condition is not useful; `field` always exists, and there is a test on `field.search` below.